transformer_engine/pytorch/csrc/comm_gemm_overlap.h

```diff
@@ -233,8 +236,6 @@ struct UbufCommOverlap : torch::CustomClassHolder, UbufBase {
     cudaEventDestroy(_stop_compute);
     cudaEventDestroy(_start_compute);
 
-    for (size_t i = 0; i < _stream_compute.size(); i++) cudaStreamDestroy(_stream_compute[i]);
```

Comment:
Are we intentionally leaking these `cudaStream_t`s? I see that PyTorch leaks intentionally because it can't guarantee that the streams are destroyed before the CUDA environment is destroyed: https://github.com/pytorch/pytorch/blob/cf11fc0dcbb9c907cf6e851109b92f4157e445c9/c10/cuda/CUDAStream.cpp#L27-L39